controllers/storagecluster/external_resources.go

```diff
@@ -326,6 +345,81 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	for _, scc := range getDefaultStorageClassClaims(instance) {
+
+		err := controllerutil.SetOwnerReference(instance, scc, r.Client.Scheme())
+		if err != nil {
+			return err
+		}
+
+		err = r.Client.Create(context.TODO(), scc)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) deleteDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	for _, scc := range getDefaultStorageClassClaims(instance) {
+
+		err := r.Client.Delete(context.TODO(), scc)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) verifyNoStorageClassClaimsExist(instance *ocsv1.StorageCluster) error {
+
+	storageClassClaims := &ocsv1alpha1.StorageClassClaimList{}
+	err := r.Client.List(context.TODO(), storageClassClaims, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(storageClassClaims.Items) != 0 {
+		err = inbuilterrors.New("Failed to cleanup resources. storageClassClaims are present." +
+			"Delete all storageClassClaims for the cleanup to proceed")
+		r.recorder.ReportIfNotPresent(instance, corev1.EventTypeWarning, "Cleanup", err.Error())
+		r.Log.Error(err, "Waiting for all storageClassClaims to be deleted.")
+		return err
+	}
+
+	return nil
+}
+
+func getDefaultStorageClassClaims(instance *ocsv1.StorageCluster) []*ocsv1alpha1.StorageClassClaim {
+
+	StorageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForDefaultStorageClassClaimFile(instance),
+			Namespace: instance.Namespace,
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "sharedfilesystem",
+		},
+	}
+
+	StorageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForDefaultStorageClassClaimBlock(instance),
+			Namespace: instance.Namespace,
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "blockpool",
+		},
+	}
+
+	return []*ocsv1alpha1.StorageClassClaim{StorageClassClaimFile, StorageClassClaimBlock}
+}
```

Comment:
Why do we need this function?
-  it just returns a new instance of hard-coded objects that better to be described at the call site.
-  it will return both in all invocation even if the call site need only one of them 
-  It creates and returns a full nested structure even in cases when we only care about some partial data.

Reply:
removed the func and created variables at call site only